### PR TITLE
Update object-storage-proxy.md

### PR DESCRIPTION
**Pull request by tomakun**
This documentation has I believe one confusing point: It uses two times `YOUR_S3_HOSTNAME` in the nginx configuration example, however the second time it looks like you actually have to use `YOUR_BUCKET_NAME.YOUR_S3_HOSTNAME`.

## Explanation
Mastodon seems to be using path-style requests to S3 which looks like this:
https://s3.region-code.amazonaws.com/bucket-name/key-name

Once you enable the S3_ALIAS_HOST in .env.production and the new nginx vhost file described in this documentation, the path-style request which goes to amazon looks like this:
https://s3.region-code.amazonaws.com/key-name

In other words, **the bucket name is not being sent to S3** and the images break.

Because of this this precedent in the file: `set $s3_backend 'https://YOUR_BUCKET_NAME.YOUR_S3_HOSTNAME';`, the current instructions make it seem like the header Host value should not include the bucket:
`proxy_set_header Host YOUR_S3_HOSTNAME;`

However it seems that for this to work with the current path-style request integration, one must use the bucket name in the Host header value for this to work, hence the proposed change to:

`proxy_set_header Host YOUR_BUCKET_NAME.YOUR_S3_HOSTNAME;`

### CODE REVIEW

**Review by helix2301 — COMMENTED** [review]
Can you add a link from Amazon documentation with this needed change?

**Comment by tomakun** [comment]
Hi @helix2301 and thank you for checking on this. Here's excerpt of the [AWS documentation](https://docs.aws.amazon.com/AmazonS3/latest/userguide/VirtualHosting.html#VirtualHostingSpecifyBucket) which mentions the need to have the bucket name in the Host header:

> if the value of the Host header ends in .s3.region-code.amazonaws.com, the bucket name is the leading component of the Host header's value up to .s3.region-code.amazonaws.com. The key for the request is the Request-URI.

Having tested this myself with and without the bucket name in the host header, I can confirm my first comment. If the bucket is not in the host header the request sent to S3 by Mastodon will be like: https://s3.region-code.amazonaws.com/key-name resulting in an error, as the bucket name is not passed in the URL sent to amazon to request the asset.

**Comment by tomakun** [comment]
Any luck on this being reviewed soon?

**Review by ClearlyClaire — APPROVED** [review]
